lib/vector-config/src/stdlib.rs

```diff
@@ -50,7 +50,7 @@ where
         // Said another way, this allows callers to use `#[configurable(derived)]` on a field of `Option<T>` so long as
         // `T` has a description, and both the optional field and the schema for `T` will get the description... but the
         // description for the optional field can still be overridden independently, etc.
-        T::metadata().convert()
+        T::metadata().convert().as_subschema()
```

Comment:
This had to change because, otherwise, we would be propagating things like default values for `T` which then makes it look like, in the schema, that any field of `Option<T>` has the defaults of `T` annotated directly on the `Option<T>` field itself.